### PR TITLE
PP-94: Revenue Sharing Model

### PR DESCRIPTION
**Pull request by mortelli**
# Revenue Sharing Model

This PR is part of the implementation of the Revenue Sharing model feature.

## Business Logic overview

The Revenue Sharing Model is a way to relay transactions so that the resulting relay fees are shared among multiple partners. 

A new type of contract (called "Collector") can now be used in relay requests, and will act as the receiver of its related relay fees. The usual receivers of these fees are the relay workers, and this will continue to be the case if a Collector is not used. This is, in fact, the way to opt out of the Revenue Sharing feature. 

At a later (and arbitrary) point in time, collected fees can be withdrawn from the contract and distributed between the different partners involved (also called "beneficiaries") which are addresses written in the contract, each with their own percentage of shares.  

For more in-depth information please see [the devportal Revenue Sharing documentation](https://github.com/mortelli/devportal/pull/20). 

## Technical overview
1. The collector contract field has been added to tests.
2. A fix for the scripts that runs the tests has been applied, in which a parameter was used in place of another one.
3. A fix for a missing folder error on Mac has been applied.
4. The `package-lock.json` file has been updated.

### Other Revenue Sharing PRs
1. `rif-relay-contracts`: [PR #22 (Revenue Sharing Model)](https://github.com/anarancio/rif-relay-contracts/pull/22)
2. `rif-relay-common`: [PR #10 (Revenue Sharing Model)]( https://github.com/infuy/rif-relay-common/pull/10)
3. `rif-relay-client`: [PR #12 (Revenue Sharing Model)](https://github.com/infuy/rif-relay-client/pull/12)
4. `rif-relay-server`: [PR #16 (Revenue Sharing Model)](https://github.com/infuy/rif-relay-server/pull/16)
5. `relaying-services-sdk`: [PR #14 (Revenue Sharing Model)](https://github.com/infuy/relaying-services-sdk/pull/14)
6. `relaying-servives-sdk-dapp`: [PP #13 (Revenue Sharing Model)](https://github.com/antomor/relaying-services-sdk-dapp/pull/13)
7. `devportal`: [Revenue Sharing documentation](https://github.com/mortelli/devportal/pull/20)

### Revenue Sharing Jira issues
See [here](https://rsklabs.atlassian.net/issues/?jql=labels%20%3D%20%22revenue-sharing%22).

### Other 
[RIF Relay - Revenue Sharing implementation](https://docs.google.com/document/d/1lmh78Bv7my5z6gkLEHuSqAj4NIQxZw6JwwfFT6zd9Dw)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -39,10 +39,10 @@
...
     "@openeth/truffle-typings": "0.0.6",
     "@openzeppelin/contracts": "^3.2.0",
-    "@rsksmart/rif-relay-client": "https://github.com/infuy/rif-relay-client",
-    "@rsksmart/rif-relay-common": "https://github.com/infuy/rif-relay-common",
-    "@rsksmart/rif-relay-contracts": "https://github.com/anarancio/rif-relay-contracts",
-    "@rsksmart/rif-relay-server": "https://github.com/infuy/rif-relay-server",
+    "@rsksmart/rif-relay-client": "https://github.com/infuy/rif-relay-client#PP-94/revenue-sharing-model",
+    "@rsksmart/rif-relay-common": "https://github.com/infuy/rif-relay-common#PP-94/revenue-sharing-model",
+    "@rsksmart/rif-relay-contracts": "https://github.com/anarancio/rif-relay-contracts#PP-94/revenue-sharing-model",
+    "@rsksmart/rif-relay-server": "https://github.com/infuy/rif-relay-server#PP-94/revenue-sharing-model",
```
Comment (mortelli) [inline package.json:45]: need to be updated once the contracts, common, client and server PRs are merged

**Comment by mortelli** [comment]
there's a conflict with the package-json file what needs to be solved

**Comment by mortelli** [comment]
requesting review from @antomor

**Comment by antomor** [comment]
This PR depends on https://github.com/rsksmart/rif-relay-contracts/pull/85. After merging the contracts PR, we need to update `package.json` and `package-lock.json`

**Pull request merged**